bin/resource_management/resource-broken-links-csv.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+import pandas as pd
+import urllib
+import sys
+from sys import stderr, stdout, argv
+from multiprocessing import Process
+
+
+# In[ ]:
+
+
+def run_cpu_tasks_in_parallel(tasks):
+    """
+    Manages processes
+    tasks - list of methods to be run
+    """
+    running_tasks = [Process(target=task) for task in tasks]
+    for running_task in running_tasks:
+        running_task.start()
+    for running_task in running_tasks:
+        running_task.join()
+        
+def main(filename,processes=4):
+    """
+    Scan url's of resources from dataset metadata obtained from https://open.canada.ca/data/en
+    CKAN API - ckanapi dump datasets --all -O datasets.jsonl.gz -z -p 4 -r http://localhost
+    
+    filename - destination of jsonl/jl file 
+    processes - default processes=4. Number of batches and processes to run. Will export 1 file per process.
+    
+    Exports broken links found in metadata
+    """
+    try:
+        df=pd.read_json(filename,lines=True)
+    except:
+        print('Error: .jsonl/.jl file expected')
+        return False
+    file_length = len(df)
+    batchsize=round(file_length/processes)
```

Comment:
if this rounds down you're going to skip as many as (processes-1) lines from the end of the file